Library/ViewModels/ProjectPamphletContentViewModelTests.swift

```diff
@@ -390,4 +399,144 @@ final class ProjectPamphletContentViewModelTests: TestCase {
+
+      // Simulate a new version of the project coming through
+      self.vm.inputs.configureWith(value: (project, .discovery))
+
+      self.loadProjectIntoDataSourceCreatorDetailsIsLoading.assertValues([true, true])
+      self.loadProjectIntoDataSourceCreatorDetailsEnvelope.assertValues([nil, nil])
+      self.loadProjectIntoDataSourceProject.assertValues(
+        [project, project], "The new project is loaded into data source"
+      )
+      self.loadProjectIntoDataSourceRefTag.assertValues(
+        [.discovery, .discovery], "The new refTag is loaded into data source"
+      )
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing new emits when the view is done.")
+
+      self.scheduler.advance()
+
+      self.loadProjectIntoDataSourceCreatorDetailsIsLoading.assertValues(
+        [true, true, false], "isLoading is false"
+      )
+      self.loadProjectIntoDataSourceCreatorDetailsEnvelope.assertValues(
+        [nil, nil, envelope], "ProjectCreatorDetailsEnvelope is returned"
+      )
+      self.loadProjectIntoDataSourceProject.assertValues([project, project, project])
+      self.loadProjectIntoDataSourceRefTag.assertValues([.discovery, .discovery, .discovery])
+      self.loadMinimalProjectIntoDataSource.assertValues([])
+    }
+  }
+
+  func testLoadProjectIntoDataSource_CreatorDetailsFailure() {
+    let project = Project.template
+
+    let mockService = MockService(fetchProjectCreatorDetailsResult: .failure(.invalidInput))
+
+    withEnvironment(apiService: mockService) {
+      self.vm.inputs.configureWith(value: (project, .discovery))
+      self.vm.inputs.viewDidLoad()
+
+      self.loadProjectIntoDataSourceCreatorDetailsIsLoading.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceCreatorDetailsEnvelope.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceProject.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceRefTag.assertValues([], "Nothing emits immediately.")
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing emits immediately.")
+
+      // Begin presentation. When presenting the project `animated` will be false since it is embedded in the
+      // navigator controller.
+      self.vm.inputs.viewWillAppear(animated: false)
+      self.vm.inputs.viewDidAppear(animated: false)
+
+      self.loadProjectIntoDataSourceCreatorDetailsIsLoading.assertValues([true], "Starts loading.")
+      self.loadProjectIntoDataSourceCreatorDetailsEnvelope.assertValues([nil], "Emits nil as view appears")
+      self.loadProjectIntoDataSourceProject
+        .assertValues([project], "Load the full project into the data source.")
+      self.loadProjectIntoDataSourceRefTag.assertValues([.discovery], "Load the refTag into the data source.")
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Do not load the minimal version of the project.")
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/kickstarter/ios-oss/blob/a8f032cddfba991aba476d145bc2cbcf057daea7/Library/ViewModels/ProjectPamphletContentViewModelTests.swift#L501"></span>Line should be 110 characters or less: currently 111 characters<br />
<code>line_length</code> <code>ProjectPamphletContentViewModelTests.swift:501</code></td>
    </tr>
  </tbody>
</table>